esmvaltool/utils/recipe_filler.py

```diff
@@ -0,0 +1,693 @@
+"""
+Fill in a blank recipe with additional datasets.
+
+Tool to obtain a set of additional datasets when given a blank recipe.
+The blank recipe should contain, to the very least, a list of diagnostics
+each with their variable(s). Example of minimum settings:
+
+diagnostics:
+  diagnostic:
+    variables:
+      ta:
+        mip: Amon
+        start_year: 1850
+        end_year: 1900
+
+Note that the tool will exit if any of these minimum settings are missing!
+
+Key features:
+
+- you can add as many variable parameters as are needed; if not added, the
+  tool will use the "*" wildcard and find all available combinations;
+- you can restrict the number of datasets to be looked for with the `dataset:`
+  key for each variable, pass a list of datasets as value, e.g.
+  `dataset: [MPI-ESM1-2-LR, MPI-ESM-LR]`;
+- you can specify a pair of experiments eg `exp: [rcp26, rcp85]`
+  for each variable; this will look for each available dataset per experiment
+  and assemble an aggregated data stretch from each experiment; equivalent to
+  esmvaltool's syntax of multiple experiments; this option needs an ensemble
+  to be declared explicitly; it will return no entry if there are gaps in data
+- `start_year` and `end_year` are mandatory and are used to filter out the
+  datasets that don't have data in the interval; if you want all possible years
+  hence no filtering on years just use "*" for start and end years;
+- `config-user: rootpath: CMIPX` may be a list, rootpath lists are supported;
+
+Caveats:
+
+- the tool doesn't yet work for derived variables;
+- operation restricted to CMIP data.
+
+Have fun!
+"""
+import argparse
+import itertools
+import logging
+import os
+import shutil
+from glob import glob
+
+import yaml
+from ruamel.yaml import YAML
+from esmvalcore._config import (configure_logging, read_config_developer_file,
```

Comment:
Please stop importing private functions from esmvalcore, I just fixed this for the CMORizer script and now you're adding new code that does this again :crying_cat_face:

Maybe you can just use [logging.basicconfig](https://docs.python.org/3/howto/logging.html#changing-the-format-of-displayed-messages) to set up logging? Logging to file seems a bit overkill to me for an interactive tool like this.

The other functions you can just copy/paste and remove the bits that aren't used here anyway, it looks like you're mostly using config-developer, right

Reply:
hahah, good point Bouwe, sozz - I'll fix that :+1:

Reply:
Thanks!